diff --git a/extensions/common/extension_resource.cc b/extensions/common/extension_resource.cc
index 1bea1813f2a5b..31172b24db6fd 100644
--- a/extensions/common/extension_resource.cc
+++ b/extensions/common/extension_resource.cc
@@ -5,20 +5,10 @@
 #include "extensions/common/extension_resource.h"
 
 #include "base/check.h"
-#include "base/feature_list.h"
 #include "base/files/file_util.h"
 
 namespace extensions {
 
-namespace {
-
-#if BUILDFLAG(IS_WIN)
-BASE_FEATURE(WindowsNormalizeFilePathFallbackOnError,
-             base::FEATURE_DISABLED_BY_DEFAULT);
-#endif
-
-}  // namespace
-
 ExtensionResource::ExtensionResource() : follow_symlinks_anywhere_(false) {}
 
 ExtensionResource::ExtensionResource(const ExtensionId& extension_id,
@@ -71,13 +60,6 @@ base::FilePath ExtensionResource::GetFilePath(
   base::FilePath normalized_extension_root;
   if (!base::NormalizeFilePath(extension_root, &normalized_extension_root)) {
 #if BUILDFLAG(IS_WIN)
-    // TODO(crbug.com/410059474): Remove this if-check and the Windows-specific
-    // fallback logic in M143.
-    if (!base::FeatureList::IsEnabled(
-            kWindowsNormalizeFilePathFallbackOnError)) {
-      return base::FilePath();
-    }
-
     // On Windows, `NormalizeFilePath` fails if the path doesn't start with a
     // drive letter (e.g. a network path) or if it exceeds `MAX_PATH` characters
     // in length. Fall back to `MakeAbsoluteFilePath` and proceed if the path
